datafusion/core/benches/merge.rs

```diff
@@ -214,6 +266,26 @@ impl MergeBenchCase {
         }
     }
 
+    fn new_with_sort_input(partitions: &[Vec<RecordBatch>]) -> Self {
+        let runtime = tokio::runtime::Builder::new_multi_thread().build().unwrap();
+        let session_ctx = SessionContext::new();
+        let task_ctx = session_ctx.task_ctx();
+
+        let schema = partitions[0][0].schema();
+        let sort = make_sort_exprs(schema.as_ref());
+
+        let projection = None;
+        let exec = Arc::new(MemoryExec::try_new(partitions, schema, projection).unwrap());
```

Comment:
I double checked that the sort exec does actually make multiple output streams. 👍 

FYI @mustafasrepo  and @ozankabak and @mingmwang   -- not sure if you are following this work from @jaylmiller but I think his work may make repartitioned sorts (in parallel) a more compelling option

Reply:
Yes, I have been following along, great work!

Reply:
Is the preserve partitioning case necessary for the SortExec bench? I'm realizing now it doesn't really measure anything useful in terms of perf 😅 thinking it might make sense to remove those cases in this PR. What do u think?@alamb